jujugui/static/gui/src/app/components/deployment-flow/sshkey/sshkey.js

```diff
@@ -30,22 +15,298 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 const AZURE_CLOUD_TYPE = 'azure';
 
 class DeploymentSSHKey extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      addSource: 'github',
+      SSHkeys: [],
+      error: null,
+      buttonDisabled: true
+    };
+  }
+
   /**
-    Handle SSH key content changes.
+    Handle keyboard input to listen for return key.
 
-    @method _onSSHKeyInputBlur
-    @param {Object} evt The blur event.
+    @param {Object} evt The keyboard event.
   */
-  _onSSHKeyInputBlur(evt) {
-    const key = this.refs.sshKey.getValue();
-    this.props.setSSHKey(key);
+  _onKeyUp(evt) {
+    if (evt.which === 13) {
+      this._handleAddMoreKeys(null);
+    }
+    const hasValue = (this.refs.sshKey && this.refs.sshKey.getValue()) ||
+      (this.refs.githubUsername && this.refs.githubUsername.getValue());
+    this.setState({
+      buttonDisabled: hasValue ? false : true
+    });
   }
 
   /**
-    Render the component.
+    Split a Manual Key into its parts.
 
-    @method render
+    @param {String} sshkey
+   */
+  _splitKey(sshKey) {
+    if (sshKey.indexOf(' ') > -1) {
+      const splitKey = sshKey.split(' ');
+      if (splitKey.length >= 2) {
+        return {
+          'body': splitKey[1],
+          'text': sshKey,
+          'type': splitKey[0],
+          'id': 0 // To conform to the schema of the github api
+        };
+      }
+    }
+    this.setState({error: (<span>Key is invalid. Please
+    ensure you have copied the key correctly.</span>)});
+    return false;
+  }
+
+  /**
+    Check if an sshkey exists to avoid duplication.
+
+    @param {Object} key The key object to check.
+    @return {Boolean}
   */
+  _keyExists(key) {
+    return this.state.SSHkeys.
+      findIndex(knownKey => knownKey.body === key.body) !== -1;
+  }
+
+  /**
+    Handle adding github username.
+
+    @param {String} error The error returned.
+    @param {Array} keys The returned keys.
+  */
+  _addGithubKeysCallback(error, keys) {
+    if (error) {
+      console.error(error);
+      this.setState({error: error});
+      return;
+    }
+
+    if (keys.length === 0) {
+      this.setState({error:
+        (<span>No keys found. <a className="link"
+          href="https://github.com/settings/keys" target="_blank">
+          Create an SSH Key</a>.</span>)});
+      return;
+    }
+
+    let SSHkeys = this.state.SSHkeys;
+    keys.forEach(key => {
+      if (!this._keyExists(key)) {
+        SSHkeys.push(key);
+      }
+    });
+    this.props.setSSHKey(SSHkeys[0].text);
```

Comment:
Yeah, @makyo is picking this up to send all the keys in a separate branch - this is just temporary